src/main/java/org/molgenis/vcf/decisiontree/filter/VcfRecord.java

```diff
@@ -67,21 +71,114 @@ public Object getValue(Field field, Allele allele, Integer sampleIndex) {
+        value = getSampleValue(field, sampleContext);
         break;
       default:
         throw new UnexpectedEnumException(fieldType);
     }
     return value;
   }
 
-  private Object getFormatField(Field field, Integer sampleIndex) {
-    Genotype genotype = variantContext.getGenotype(sampleIndex);
+  private Object getSampleValue(Field field, SampleContext sampleContext) {
+    Object value;
+    switch (field.getId().toUpperCase()) {
+      case "AFFECTED_STATUS":
+        value = sampleContext.getAffectedStatus().toString();
+        break;
+      case "PROBAND":
+        value = sampleContext.getProband();
+        break;
+      case "SEX":
+        value = sampleContext.getSex().toString();
+        break;
+      case "FATHER":
+        value = sampleContext.getFather();
+        break;
+      case "MOTHER":
+        value = sampleContext.getMother();
+        break;
+      case "FAMILY":
+        value = sampleContext.getFamily();
+        break;
+      case "PHENOTYPES":
+        value = sampleContext.getPhenotypes();
+        break;
+      default:
+        throw new UnknownFieldException(field.getId(), FieldType.SAMPLE);
+    }
+    return value;
+  }
+
+  private Object getNestedGTValue(NestedField field, SampleContext sampleContext) {
+    Genotype genotype = variantContext.getGenotype(sampleContext.getIndex());
+    Object value;
+    switch (GenotypeFieldType.valueOf(field.getId())) {
+      case ALLELES:
+        value = genotype.getAlleles().stream().map(
+            htsjdk.variant.variantcontext.Allele::getBaseString).toList();
+        break;
+      case ALLELE_NUM:
+        value = genotype.getAlleles().stream().map(allele ->
+            variantContext.getAlleles().indexOf(allele)).toList();
+        break;
+      case TYPE:
+        switch (genotype.getType()) {
+          case MIXED:
+            value = GenotypeType.MIXED.name();
+            break;
+          case HET:
+            value = GenotypeType.HET.name();
+            break;
+          case HOM_REF:
+            value = GenotypeType.HOM_REF.name();
+            break;
+          case HOM_VAR:
+            value = GenotypeType.HOM_VAR.name();
+            break;
+          case NO_CALL:
+            value = GenotypeType.NO_CALL.name();
+            break;
+          case UNAVAILABLE:
+            value = GenotypeType.UNAVAILABLE.name();
+            break;
+          default:
+            throw new UnexpectedEnumException(genotype.getType());
+        }
+        break;
+      case MIXED:
+        value = genotype.isMixed();
+        break;
+      case CALLED:
+        value = genotype.isCalled();
+        break;
+      case PLOIDY:
+        value = genotype.getPloidy();
+        break;
+      case PHASED:
+        value = genotype.isPhased();
+        break;
+      case NON_INFORMATIVE:
+        value = genotype.isNonInformative();
+        break;
+      default:
+        throw new UnexpectedEnumException(GenotypeFieldType.valueOf(field.getId()));
```

Comment:
dedup: store `switch (GenotypeFieldType.valueOf(field.getId()))` in var and use here